### ISSUE TITLE
Visualizer2 download error

### ISSUE THREAD

**Issue by govertbuijs** [powertac/powertac-server#991]
While trying to download a file from the web interface, the following error occurs:

{"message":"error.internalServerError","description":"Internal server error: could not initialize proxy [org.powertac.visualizer.domain.File#3] - no Session","fieldErrors":null}

**Comment by jecollins** [comment]
I also observe this behavior. Is this a result of removing a cookie?

**Comment by jecollins** [comment]
Perhaps this is a different issue, but I did a pull and build, and now I get an error from viz2 on startup, using mvn -Pweb2 in server-distribution:
```
2018-11-22 09:26:47,005 main WARN Error while converting string [#log4j2.loglevel#] to type [class org.apache.logging.log4j.Level]. Using default value [null]. java.lang.IllegalArgumentException: Unknown level constant [#LOG4J2.LOGLEVEL#].
	at org.apache.logging.log4j.Level.valueOf(Level.java:320)
	at org.apache.logging.log4j.core.config.plugins.convert.TypeConverters$LevelConverter.convert(TypeConverters.java:288)
	at org.apache.logging.log4j.core.config.plugins.convert.TypeConverters$LevelConverter.convert(TypeConverters.java:284)
	at org.apache.logging.log4j.core.config.plugins.convert.TypeConverters.convert(TypeConverters.java:419)
	at org.apache.logging.log4j.core.config.plugins.visitors.AbstractPluginVisitor.convert(AbstractPluginVisitor.java:149)
	at org.apache.logging.log4j.core.config.plugins.visitors.PluginAttributeVisitor.visit(PluginAttributeVisitor.java:45)
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.generateParameters(PluginBuilder.java:254)
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.build(PluginBuilder.java:136)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createPluginObject(AbstractConfiguration.java:959)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:899)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:891)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.doConfigure(AbstractConfiguration.java:514)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.initialize(AbstractConfiguration.java:238)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.start(AbstractConfiguration.java:250)
	at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:547)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:619)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:636)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:231)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:153)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:45)
	at org.apache.logging.log4j.LogManager.getContext(LogManager.java:194)
	at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getContext(AbstractLoggerAdapter.java:121)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getContext(Log4jLoggerFactory.java:43)
	at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getLogger(AbstractLoggerAdapter.java:46)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:29)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:358)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:383)
	at org.powertac.visualizer.Visualizer2App.<clinit>(Visualizer2App.java:37)
```

**Comment by erikkemperman** [comment]
Strange. That is a placeholder which should have been replaced by maven. I did the following, and it seems to work:

- checkout latest master branch of powertac-server from github
- `mvn install` in the visualizer2 directory
- `cd path/to/server-distribution`
- `rm -rf target` (to get rid of earlier versions)
- `mvn -Pweb2`

To check, you can open the built war file and examine the entry for `WEB-INF/classes/log4j2.xml` (you can open the war file in any zip/archive viewer, it's just ZIP). The original version has the placeholders, e.g. here:
https://github.com/powertac/powertac-server/blob/master/visualizer2/src/main/resources/log4j2.xml#L45

But the version deployed as a part of the war should have these replaced by actual log levels(depending on the profile, `dev` vs `prod` you get different log levels).

**Comment by jecollins** [comment]
Very strange. I tried just cleaning server-distribution/target (using mvn clean) but that was not enough. Note that I had done a clean install in powertac-server just before the failure. So I did another clean install in powertac-server, and a clean in server-distribution, then it worked. I'll do a new deploy and test, then we may have to tell folks that they may have to do a clean in server-distribution if they run into this problem.
